Plugins/SwiftProtobufPlugin/plugin.swift

```diff
@@ -0,0 +1,139 @@
+import Foundation
+import PackagePlugin
+
+@main
+struct SwiftProtobufPlugin: BuildToolPlugin {
+    /// Errors thrown by the `SwiftProtobufPlugin`
+    enum PluginError: Error {
+        /// Indicates that the target where the plugin was applied to was not `SourceModuleTarget`.
+        case invalidTarget
+        /// Indicates that no config file was present in the root of the target directory.
+        case noConfigFile
+    }
+
+    /// The configuration of the plugin.
+    struct Configuration: Codable {
+        /// Encapsulates a single invocation of protoc.
+        struct Invocation: Codable {
+            /// The visibility of the generated files.
+            enum Visibility: String, Codable {
+                /// The generated files should have `internal` access level.
+                case `internal`
+                /// The generated files should have `public` access level.
+                case `public`
+            }
+
+            /// An array of paths to `.proto` files for this invocation.
+            var protoFiles: [String]
+            /// The visibility of the generated files.
+            var visibility: Visibility?
+        }
+
+        /// The path to the `protoc` binary.
+        ///
+        /// If this is not set, SPM will try to find the tool itself.
+        var protocPath: String?
```

Comment:
`protocPath` is fundamentally different from the rest of the configuration options here:  It's a property of the build environment, not the project.

I've seen other build tools allow for two configuration files:  One that is checked in and holds configuration that is independent of the particular build environment and a second one that is typically _not_ checked in with configuration for the build environment.  In this case, having a second "swift-protobuf-config-local.json" that will override the protocPath if present might be a good approach.

Reply:
I really like that idea! Let me change that up and separate it into two files.

Reply:
I would still allow `protocPath` in the checked-in file (for use of CI systems that may not be able to easily manage local configuration files).  But the `-local` file should override that setting.

Reply:
@tbkka Friendly ping. Would love to get your input on this once more.

Reply:
You've convinced me that we don't need local configuration support.